src/App/Fossa/Config/ConfigFile.hs

```diff
@@ -90,18 +94,23 @@ resolveConfigFile base path = do
   loc <- resolveLocation base path
   case loc of
     DefaultConfigLocation dir -> do
-      let realpath = dir </> defaultConfigFileName
-      exists <- doesFileExist realpath
-      if not exists
-        then -- File missing and not requested
-          pure Nothing
-        else do
-          configFile <- readContentsYaml realpath
+      possibleConfigFilePaths <-
+        mapM
+          ( \configFileName -> do
+              let configFilePath = dir </> configFileName
+              exists <- doesFileExist configFilePath
+              pure $ if exists then Just configFilePath else Nothing
+          )
+          defaultConfigFileNames
+      case asum possibleConfigFilePaths of
+        Just actualConfigFilePath -> do
+          configFile <- readContentsYaml actualConfigFilePath
           let version = configVersion configFile
           if version >= 3
             then pure $ Just configFile
             else -- Invalid config found without --config flag: warn and ignore file.
               logWarn (warnMsgForOlderConfig @AnsiStyle version) $> Nothing
+        Nothing -> pure Nothing
```

Comment:
When I see this in a case statement, I'm usually pretty sure we can just use the monad instead (if there's no `pure`, we can use the functor). something like `asum possibleConfigFilePaths >>= do ...`.  I don't know if that's more readable, but I generally prefer that to case statements.

Reply:
I've rewritten this using `whenJust`. Does this feel better? Should I refactor `whenJust` into a helper somewhere?

We can't exactly use the `Maybe` monad here because the inner action is itself effectful.

Reply:
Damn, you're right.  I did say that it was a very quick look.  I prefer it the previous way for now.

Reply:
Yeah, I agree - it's annoying but clearer. Will revert and then merge.